arkouda/pdarraysetops.py

```diff
@@ -432,30 +488,54 @@ def setdiff1d(pda1: pdarray, pda2: pdarray,
 
     Examples
     --------
+    Singular pdarrays
     >>> a = ak.array([1, 2, 3, 2, 4, 1])
     >>> b = ak.array([3, 4, 5, 6])
     >>> ak.setdiff1d(a, b)
     array([1, 2])
+
+    Multiple pdarrays flattened
+    a = [0, 1, 2]
+        b = [3, 4, 5]
+        c = [1]
+        d = [3, 4]
+    >>> a_segs = ak.array([0, 3])
+    >>> a_vals = ak.array([0, 1, 2, 3, 4, 5])
+    >>> b_segs = ak.array([0, 2])
+    >>> b_vals = ak.array([1, 3, 4])
+    >>> ak.setdiff1d((a_segs, a_vals), (b_segs, b_vals))
+    (array([0, 2]), array(0, 2, 5))
     """
-    if pda1.size == 0:
-        return pda1  # return a zero length pdarray
-    if pda2.size == 0:
-        return pda1  # subtracting nothing return orig pdarray
-    if (pda1.dtype == int and pda2.dtype == int) or \
-            (pda1.dtype == akuint64 and pda2.dtype == akuint64):
-        repMsg = generic_msg(cmd="setdiff1d", args="{} {} {}". \
-                             format(pda1.name, pda2.name, assume_unique))
-        return create_pdarray(cast(str, repMsg))
-    if not assume_unique:
-        pda1 = cast(pdarray, unique(pda1))
-        pda2 = cast(pdarray, unique(pda2))
-    return pda1[in1d(pda1, pda2, invert=True)]
+    if isinstance(pda1, pdarray) or isinstance(pda2, pdarray):
```

Comment:
as suggested below, recommend helper methods for checking if two params are of the same type, could even add 'em as a decorator